src/main/java/at/ac/tuwien/kr/alpha/common/depgraph/Node.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2019, the Alpha Team.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1) Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2) Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package at.ac.tuwien.kr.alpha.common.depgraph;
+
+import at.ac.tuwien.kr.alpha.common.Predicate;
+
+/**
+ * A node in a dependency graph. One node references exactly one predicate. This means that all rule heads deriving the same predicate will be condensed into
+ * the same graph node. In some cases this results in more "conservative" results in stratification analysis, where some rules will not be evaluated up-front,
+ * although that would be possible.
+ * 
+ * Copyright (c) 2017-2019, the Alpha Team.
+ */
+public class Node {
+
+	private final Predicate predicate;
+	private final String label;
+	private final boolean isConstraint;
```

Comment:
True, I thought I'd need it and then never did.. I'd still like to keep it since it's probably going to be used in the future. An example I could think of is extending stratified evaluation to deal with constraints that can be fully grounded - currently the stratified evaluator doesn't do anything with constraints, but it would be an easy to implement feature for which that information would be needed.
Another use case for this flag I can think of is if one wanted to "fancy up" the dot files written for dependency graphs and use a different color for graph nodes representing contraints.